File: backend/src/main/scala/kpn/api/common/network/NetworkChangesPage.scala
package kpn.api.common.network

import kpn.api.common.changes.details.NetworkChangeInfo
import kpn.api.common.changes.filter.ChangesFilterOption

case class NetworkChangesPage(
  network: NetworkSummary,
  filterOptions: Seq[ChangesFilterOption],
  changes: Seq[NetworkChangeInfo],
  totalCount: Long
)

File: backend/api/src/main/java/kpn/api/common/network/NetworkChangesPage.java
package kpn.api.common.network;

import kpn.api.common.changes.details.NetworkChangeInfo;
import kpn.api.common.changes.filter.ChangesFilterOption;
import kpn.api.common.network.NetworkSummary;

import com.google.common.collect.ImmutableList;

public record NetworkChangesPage(
  NetworkSummary network,
  ImmutableList<ChangesFilterOption> filterOptions,
  ImmutableList<NetworkChangeInfo> changes,
  Long totalCount
) {
}
